tests/run-make/pgo-branch-weights/rmake.rs

```diff
@@ -0,0 +1,75 @@
+// This test generates an instrumented binary - a program which
+// will keep track of how many times it calls each function, a useful
+// feature for optimization. Then, an argument (aaaaaaaaaaaa2bbbbbbbbbbbb2bbbbbbbbbbbbbbbbcc)
+// is passed into the instrumented binary, which should react with a number of function calls
+// fully known in advance. (For example, the letter 'a' results in calling f1())
+
+// If the test passes, the expected function call count was added to the use-phase LLVM-IR.
+// See https://github.com/rust-lang/rust/pull/66631
+
+// FIXME(mati865): MinGW GCC miscompiles compiler-rt profiling library but with Clang it works
+// properly. Since we only have GCC on the CI ignore the test for now.
+
+//@ needs-profiler-support
+//@ ignore-windows-gnu
+//@ ignore-cross-compile
+
+use std::io::{self, Write};
+use std::process::Command;
+use run_make_support::{llvm_profdata, rustc, rustdoc, tmp_dir};
+
+fn main() {
+    // For some very small programs GNU ld seems to not properly handle
+    // instrumentation sections correctly. Neither Gold nor LLD have that
+    // problem.use run_make_support::{run, rustc, tmp_dir};
+    let common_flags = if env::consts::OS == "linux"
```

Comment:
I think it's better if this is called `link_args` and is an `Option<&'static str>`, so that it is only added to a `rustc()` invocation if it is `Some` (so we don't pass empty `""` arguments):

```rs
let link_args = (cfg!(target_os = "linux") && target().contains("x86")).then_some("-Clink-args=-fuse-ld=gold");

let mut invoc = rustc()
    // ... some other args()
    ;

if let Some(link_args) = link_args {
    invoc.arg(link_args);
}
```

Reply:
Added, though as per your other review comment, it may be interesting to delete this check entirely and see what happens.